### PR TITLE
Added properties to collections

### PR DESCRIPTION
**Pull request by thehorse2000**
# Description

This change introduces storing properties for collections.
Currently, two properties are stored in collections:
1. BASE URL which will be the initial default value for any new request within that collection
2. Default Request Type also sets initially the request type for new requests.

This is very helpful to avoid repetition of entering the same URL over and over again.
This also opens up more space for future customizations for the collections.

#### 
The collection properties are stored in the bruno.json file of the collection
### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**

For previous comments of this PR. See https://github.com/usebruno/bruno/pull/707

image

https://github.com/usebruno/bruno/assets/7264374/2cf9516a-36e2-428a-a0c0-522558bb7da2

### CODE REVIEW

**Comment by helloanoop** [comment]
@thehorse2000 Nice!

Fully onboard with supporting this. I

I'd like to recommend a few changes. 
The `properties` modal was kind of inspired by how we have properties of a file or folder in any OS.
I recommend putting this inside the collection settings and calling the Tab `Presets` 
The config can be stored inside `bruno.json` under the key `presets`
```json
{
  "presets": {
     "defaultUrl": "",
     "defaultRequestType": "",
  }
}
```

image

**Comment by thehorse2000** [comment]
Hmm, I understand your view. But don't you think it would be more noticeable if we insert a new tab item for collections named Presets and introduce a new modal for it?
It will contain those two fields, And more hopefully in the future.
My concern is that it might not be noticeable at all if we set it in collection settings. Because I noticed that we can only access the collection settings from the request page. I didn't even notice it before you mentioned it 😢  

What do you think?
image

**Comment by thehorse2000** [comment]
Hey, Thanks for the info. I actually didn't know there was a collections settings area 😿 
I think that is because we can only access it from requests.
I believe it would be better if we could access these settings from the menu of the collection (like properties)
But maybe that is for another PR 😄

**Comment by thehorse2000** [comment]
### New Changes
- Changed "properties" to "presets" in bruno.json file
- Added Presets tab to collection settings
- Returned collection properties modal to its previous state.

image

https://github.com/usebruno/bruno/assets/7264374/39dd48e7-510a-4988-b662-42661eaff65a

File: packages/bruno-electron/src/ipc/collection.js [hunk 1]
```diff
@@ -94,6 +94,28 @@ const registerRendererEventHandlers = (mainWindow, watcher, lastOpenedCollection
     }
   });
 
+  // update collection properties
```
Comment (helloanoop) [inline packages/bruno-electron/src/ipc/collection.js]: @thehorse2000 You don't need a new electron event. There is already one which handles the updates for Bruno Config file.

https://github.com/usebruno/bruno/blob/main/packages/bruno-app/src/components/CollectionSettings/index.js#L34-L42

Can you update the code to use that?

File: packages/bruno-app/src/components/CollectionSettings/index.js [hunk 2]
```diff
@@ -114,6 +118,9 @@ const CollectionSettings = ({ collection }) => {
         <div className={getTabClassname('headers')} role="tab" onClick={() => setTab('headers')}>
           Headers
         </div>
+        <div className={getTabClassname('presets')} role="tab" onClick={() => setTab('presets')}>
```
Comment (helloanoop) [inline packages/bruno-app/src/components/CollectionSettings/index.js]: Lets place this tab after `Tests` and before `Proxy`

**Review by helloanoop — CHANGES_REQUESTED** [review]
@thehorse2000 Nice work. I have a couple more comments. Once these addressed, I feel this should be good to merge.

**Comment by martinsefcik** [comment]
@thehorse2000 @helloanoop Probably this Settings tab for collection could be opened by clicking on collection name in sidebar, the same way how it is working in Postman. It could be then also used to switch quickly environment, because now if I want to run whole collection or folder I have to unfold many subfolders find first request in collection open it and then I can switch environment which is not very user friendly.

Also this Collection properties in modal and Collection settings in tab could be merged to Collection tab. It doeas not make sense to me to have two places in app for similar things.

**Comment by helloanoop** [comment]
> Probably this Settings tab for collection could be opened by clicking on collection name

Agreed.

> Also this Collection properties in modal and Collection settings in tab could be merged to Collection tab. It doeas not make sense to me to have two places in app for similar things.

Agreed. Thinking of a new Tab called `Info` which will display the same stuff that we display in `Properties`

**Comment by thehorse2000** [comment]
@helloanoop Resolved the required changes.
I do also agree with you and @martinsefcik opinions on making the collections settings page more accessible.
So I will work on it, if you guys don't mind.
Will create a new PR for it and mention you

**Review by helloanoop — APPROVED** [review]

**Comment by helloanoop** [comment]
Nice work @thehorse2000 !

PR Merged.

**Comment by helloanoop** [comment]
> Probably this Settings tab for collection could be opened by clicking on collection name in sidebar

@martinsefcik This is done.

> Also this Collection properties in modal and Collection settings in tab could be merged to Collection tab. It doeas not make sense to me to have two places in app for similar things.

This is also done. See #989
I also removed `Properties` in the collection dropdown and added `Settings`

>  It could be then also used to switch quickly environment, because now if I want to run whole collection or folder I have to unfold many subfolders find first request in collection open it and then I can switch environment which is not very user friendly.

@martinsefcik I didn't understand. Can you explain this?
You should be able to select any request and switch the env on the top-right corner right ?

**Comment by martinsefcik** [comment]
> > It could be then also used to switch quickly environment, because now if I want to run whole collection or folder I have to unfold many subfolders find first request in collection open it and then I can switch environment which is not very user friendly.
> 
> @martinsefcik I didn't understand. Can you explain this? You should be able to select any request and switch the env on the top-right corner right ?

@helloanoop If you have for example the following collection structure:

```
- folder 1
  - folder 1.1
    - folder 1.1.1
      - folder 1.1.1.1
        - request 1
        - request 2
        - ...
- folder 2
  - folder 2.1
    - folder 2.1.1
      - folder 2.1.1.1
        - request a
        - request b
        - ...
```
then you have to unfold first 4 folders to get inside folder with request, then open request and the switch environment.

So I meant this that it isn't/wasn't very user friendly.

**Comment by garposmack** [comment]
I loved this new feature.
It's possible to go one step further, and have folders inside a collection to have the same settings and properties as a collection?

**Pull request merged**